lib/godmin/authentication.rb

```diff
@@ -6,26 +6,27 @@ module Authentication
     extend ActiveSupport::Concern
 
     included do
-      before_action :authenticate_admin_user
+      before_action :authenticate
 
       helper_method :admin_user
       helper_method :admin_user_signed_in?
     end
 
-    def authenticate_admin_user
-      unless admin_user_signed_in? || controller_name == "sessions"
-        redirect_to new_session_path
-      end
-    end
+    def authenticate
+      return unless authentication_enabled?
+      return if admin_user_signed_in?
+      return if controller_name == "sessions"
```

Comment:
Perhaps we should remove this and put a `prepend_before_action :disable_authentication` in the `sessions_controller` file instead? This doesn't work if the controller is named something else, as demonstrated by the `test/dummy/app/controllers/another_admin_sessions_controller.rb` file in this very PR.

Reply:
Also, perhaps we should only skip authentication when creating a new session, not destroying one

Reply:
Let's do this